pkg/pillar/zfs/zfs.go

```diff
@@ -127,11 +121,11 @@ func CreateVaultVolumeDataset(log *base.LogObject, datasetName string, zfsKeyFil
 	props[libzfs.DatasetPropVolsize] = libzfs.Property{
 		Value: strconv.FormatUint(alignedSize, 10)}
 	props[libzfs.DatasetPropVolblocksize] = libzfs.Property{
-		Value: strconv.FormatUint(VolBlockSize, 10)}
+		Value: strconv.FormatUint(blockSize, 10)}
 	props[libzfs.DatasetPropVolmode] = libzfs.Property{
 		Value: "dev"}
 	props[libzfs.DatasetPropCompression] = libzfs.Property{
-		Value: "zstd"}
+		Value: compression}
```

Comment:
What is this change about ?  I thought we use zstd compression, why remove it ?

Reply:
OK I understand the change, you are explicitly saying compression "off" but why are we turning off compression for this zvol ? Performance issues ?

Reply:
Ok can you rename that to compression_type ? In that way its clear the type we are using zstd or off. Just compression implies compression is enabled.

Reply:
how about more camel-case like compressionType to match surrounding styling?